### PR TITLE
Cosmos transaction definitions

### PR DESCRIPTION
**Pull request by patiee**
Do we want to keep all proto definitions here or create a new repository for each network?

### CODE REVIEW

File: cosmos.v1.proto
Comment (lavabyrd) [inline cosmos.v1.proto:15]: looks like the formatting is a bit off here

File: cosmos.v1 copy.proto
Comment (lavabyrd) [inline cosmos.v1 copy.proto:1]: guessing this entire file isn't needed?

File: cosmos.v1.proto
Comment (lavabyrd) [inline cosmos.v1.proto:21]: formatting

File: cosmos.v1.proto
Comment (lavabyrd) [inline cosmos.v1.proto:99]: spelling :) should be messages (just for consistency)

**Comment by lavabyrd** [comment]
actually, where is the source of the cosmos proto file coming from?

File: cosmos.v1.proto
Comment (patiee) [reply cosmos.v1.proto:15]: You're right, I need to format this file thanks!

File: codec.proto
Comment (lavabyrd) [inline codec.proto:12]: shouldn't this be 2 lines?

File: codec.proto
Comment (lavabyrd) [inline codec.proto:19]: I think the formatting broke here because of the comments

File: codec.proto
Comment (lavabyrd) [inline codec.proto:24]: single line again

File: codec.proto
Comment (lavabyrd) [inline codec.proto:35]: this seems like its not being rebased from Main. The current master already has EventBlock, can you rebase/merge it first so we can make sure we're looking at the same

File: codec.proto
Comment (lavabyrd) [inline codec.proto:49]: I don't think we need Reward anymore do we? Unless its needed for Tx Decoding

File: codec.proto
Comment (lavabyrd) [inline codec.proto:80]: broken spacing here, probably due to the comment but the `=` isn't lining up

File: codec.proto
Comment (lavabyrd) [inline codec.proto:97]: can we leave this as is?

File: codec.proto
Comment (lavabyrd) [inline codec.proto:130]: same as above

File: codec.proto
Comment (lavabyrd) [inline codec.proto:137]: weird spacing here, I get that its to match up with the `event_vote_type` but it then doesn't line up with the `validator_address`

File: codec.proto
Comment (lavabyrd) [inline codec.proto:152]: formatting

File: cosmos.v1.proto
Comment (lavabyrd) [inline cosmos.v1.proto:5]: do we build this entire file manually? Or is it a copy paste from the cosmos-sdk?

File: cosmos.v1.proto
Comment (lavabyrd) [inline cosmos.v1.proto:12]: what are these for?

File: cosmos.v1.proto
Comment (lavabyrd) [inline cosmos.v1.proto:9]: I know in codec.proto we didn't need this, may not be needed here either (it'll be removed in the TM codecs at a later date)

File: cosmos.v1.proto
Comment (lavabyrd) [inline cosmos.v1.proto:69]: is the castrepeated something we added? If so, what is it needed for?

File: cosmos.v1.proto
Comment (lavabyrd) [inline cosmos.v1.proto:482]: new line

**Review by lavabyrd — CHANGES_REQUESTED** [review]
something funny up with the rebasing/base branch here. Can we rebase/merge from main first? 

I also can't tell where the cosmos file is coming from, is it in the cosmos-sdk?

File: cosmos.v1.proto
Comment (patiee) [reply cosmos.v1.proto:5]: it is copy paste from cosmos-sdk

File: codec.proto
Comment (patiee) [reply codec.proto:19]: I'll revert formatting from this file, shouldn't touch it

File: cosmos.v1.proto
Comment (patiee) [reply cosmos.v1.proto:9]: I just checked it and this import has to be here, otherwise I got error:
`proto:463:8: "google.protobuf.MessageOptions" seems to be defined in "google/protobuf/descriptor.proto", which is not imported by "cosmos.v1.proto".  To use it here, please add the necessary import.`

File: cosmos.v1.proto
Comment (patiee) [reply cosmos.v1.proto:12]: this can be ommited, thanks

File: cosmos.v1.proto
Comment (patiee) [reply cosmos.v1.proto:69]: it's copied from cosmos-sdk, should I update castrepeated value?

File: codec.proto
Comment (patiee) [reply codec.proto]: I had to update base branch cosmos, now it looks better

**Comment by lavabyrd** [comment]
@patiee is there more changes for this or is it good to review now?

**Comment by patiee** [comment]
> @patiee is there more changes for this or is it good to review now?

Yes it's ready for a review

**Comment by sosedoff** [comment]
Im fine with keeping the cosmos proto in here

**Comment by sosedoff** [comment]
Also, if we're not planning on any new changes and everything works fine - you got a LGTM from me, lets just merge this and address any changes in separate PRs.

**Comment by lavabyrd** [comment]
can you run a `make generate` before we merge this however so we have the updated pb.go file in the codec folder too?

**Review by lavabyrd — APPROVED** [review]

**Pull request merged**
